storage.go

```diff
@@ -267,12 +287,26 @@ func (s *Storage) stat(ctx context.Context, path string, opt pairStorageStat) (o
 }
 
 func (s *Storage) write(ctx context.Context, path string, r io.Reader, size int64, opt pairStorageWrite) (n int64, err error) {
+	rp := s.getAbsPath(path)
+
+	// kodo `put` doesn't support `overwrite`, so we need to check and delete the object if exists.
+	// ref: [GSP-134](https://github.com/beyondstorage/go-storage/blob/master/docs/rfcs/134-write-behavior-consistency.md)
+	_, err = s.bucket.Stat(s.name, rp)
+	if err == nil {
+		err = s.bucket.Delete(s.name, rp)
+	}
+	if err != nil && checkError(err, responseCodeResourceNotExist) {
+		err = nil
+	}
+
+	if err != nil {
+		return
+	}
```

Comment:
```suggestion
	if err == nil {
		err = s.bucket.Delete(s.name, rp)
		if err != nil {
			return
		}
	} else if !checkError(err, responseCodeResourceNotExist) {
		return
	}
```

Will this look clearer (and equivalent)?